scraper/utils.py

```diff
@@ -0,0 +1,89 @@
+# -------------------------------------------------------------
+# Helper functions that are used once or twice & aren't
+# dependent on external variables
+# -------------------------------------------------------------
+
+import yaml
+import os
+import urllib.request
+
+# -------------------------------------------------------------
+# only used once in the scroll function
+# -------------------------------------------------------------
+def check_height():
+    new_height = driver.execute_script(selectors.get("height_script"))
+    return new_height != old_height
+
+
+# -------------------------------------------------------------
+# gets facebook credentials from the credentials.yml file.
+# -------------------------------------------------------------
+def get_credentials():
+    with open("credentials.yaml", "r") as ymlfile:
+        cfg = yaml.safe_load(stream=ymlfile)
+
+        if ("password" not in cfg) or ("email" not in cfg):
+            print(
+                "Your email or password is missing. Kindly write them in credentials.txt"
+            )
+            exit(1)
+        return cfg
+
+
+# -------------------------------------------------------------
+# takes a url and downloads image from that url
+# -------------------------------------------------------------
+def image_downloader(img_links, folder_name):
+    img_names = []
+
+    try:
+        parent = os.getcwd()
+        try:
+            folder = os.path.join(os.getcwd(), folder_name)
+            create_folder(folder)
+            os.chdir(folder)
+        except Exception:
+            print("Error in changing directory.")
+
+        for link in img_links:
+            img_name = "None"
+
+            if link != "None":
+                img_name = (link.split(".jpg")[0]).split("/")[-1] + ".jpg"
+
+                # this is the image id when there's no profile pic
+                if img_name == selectors.get("default_image"):
+                    img_name = "None"
+                else:
+                    try:
+                        req = urllib.request.Request(link)
+                        with urllib.request.urlopen(req) as response:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)